documentation/src/docs/security/securing-distributed-mode-cluster.md

```diff
@@ -0,0 +1,311 @@
+
+Setting up security - especially TLS - in a large cluster can be daunting at first. To make it easier, this document provides step-by-step instructions on how to enable security when deloying a Pravega cluster manually.
+
+Depending on the deployment option used and your environment, you might need to modify the steps and commands  to suite your specific needs and policies. Also, some of the deployment options open up additional ways of securing a cluster, as we will see in the next sub-section; We'll not discuss those ways in detail in this document, for the sake of brevity.
+
+## Setting up SSL/TLS
+
+There are two common ways of using TLS for client-server communications:
+
+1. Setup Pravega to handle TLS directly.
+2. Terminate TLS outside of Pravega, in an infrastructure component such as a reverse proxy or a Load Balancer.
+
+Depending on the deployment option used, it might be easier to use one or the other approach. For instance, if you are deploying a Pravega cluster in Kubernetes, you might find using approach 2 simpler to use and manage. Also, the specifics of how to enable TLS will differ depending on the deployment option used. This document focuses on approach 1. For approach 2, refer to the vendor's documentation.
+
+At a high level, setting up TLS in Pravega can be divided into three distinct stages:
+
+1. Setting up a Certificate Authority (CA)
+2. Obtaining server certificates and keys
+3. Deploying certificates and enabling TLS in Pravega
+
+The following sub-sections discuss the steps required for each of these phases. For setting up Zookeeper security, refer to the security documentation of Zookeeper.
+
+**Before you Begin:**
+
+You need to have the following installed on the server(s), as the commands in this section use either OpenSSL or Java Keytool.
+* OpenSSL
+* A supported version of Java Development Kit (JDK)
+
+Also, note the following:
+
+* In the examples shown in this section, we use command line arguments to pass all inputs to the command. To pass sensitive command arguments via prompts instead, just exclude the corresponding option. For example,
+
+  ```
+  # Inputs passed as command line arguments.
+  $ keytool -keystore server01.keystore.jks -alias server01 -validity <validity> -genkey \
+              -storepass <keystore-password> -keypass <key-password> \
+              -dname <distinguished-name> -ext SAN=DNS:<hostname>,
+
+  # Passwords and other arguments entered interactively on the prompt.
+  $ keytool -keystore server01.keystore.jks -alias server01 -genkey
+  ```
+* In the samples below, a weak password `changeit` is used everywhere for easier reading. Be sure to replace it with strong and separate passwords for each file.
+
+### Stage 1: Setting up a Certificate Authority (CA)
+
+If you are going to use an existing public or internal CA service or certificate and key bundle, you may skip this stage.
```

Comment:
**Note:** If you are using an existing public or internal CA service or certificate and a key bundle,  Stage 1 can be skipped.

Reply:
I have modified it a bit in a recent revision. Please check it and let me know if you think we should change it.